### ISSUE TITLE
Nicer error message when trying to access private project?

### ISSUE THREAD

**Issue by benlindsay** [dib-lab/osf-cli#77]
Sorry to raise another issue, but I really like the idea of having this command-line client and I hope the feedback is helpful. When I try to access a private project, I get a messy traceback. I see that this traceback is caused by a 401 message from osf.io. Is there a way to tell that the user is trying to access a private repo and tell that to the user rather than spitting out a traceback, or is osf.io pretty opaque about that?

```
(osf3) lindsb@rrlogin:osf-cli$ osf -p wy5pj ls
Traceback (most recent call last):
  File "/home/lindsb/usr/miniconda/envs/osf3/bin/osf", line 11, in <module>
    load_entry_point('osfclient', 'console_scripts', 'osf')()
  File "/mnt/rrio1/home/lindsb/usr/osf-cli/osfclient/__main__.py", line 97, in main
    exit_code = args.func(args)
  File "/mnt/rrio1/home/lindsb/usr/osf-cli/osfclient/cli.py", line 182, in list_
    project = osf.project(args.project)
  File "/mnt/rrio1/home/lindsb/usr/osf-cli/osfclient/api.py", line 24, in project
    return Project(self._json(self._get(url), 200), self.session)
  File "/mnt/rrio1/home/lindsb/usr/osf-cli/osfclient/models/core.py", line 23, in _get
    return self.session.get(url, *args, **kwargs)
  File "/mnt/rrio1/home/lindsb/usr/osf-cli/osfclient/models/session.py", line 45, in get
    raise UnauthorizedException()
osfclient.exceptions.UnauthorizedException
```

**Comment by betatim** [comment]
Thanks for using osfclient even though it is super new :)

No reason for the ugly traceback other than we haven't gotten to printing out a better error message. If you'd like to give implementing this a shot that would be welcome as well.

**Comment by betatim** [comment]
And another one fixed @benlindsay :smile:

**Comment by benlindsay** [comment]
Awesome! Now once the recursive file uploading pull request is merged I expect I'll start using this tool all the time.
